internal/agent/agent.go

```diff
@@ -181,20 +186,22 @@ func (c *context) IDLookup() IDLookup {
 type IDLookup map[string]string
 
 //NewContext creates a new context.
-func NewContext(cfg *config.Config, buildVersion string, resolver hostname.ResolverChangeNotifier, lookup IDLookup) *context {
+func NewContext(cfg *config.Config, buildVersion string, resolver hostname.ResolverChangeNotifier, lookup IDLookup,
+	sampleMatcher includeSampleMatcher) *context {
 	ctx, cancel := context2.WithCancel(context2.Background())
 
 	return &context{
-		cfg:            cfg,
-		Ctx:            ctx,
-		CancelFn:       cancel,
-		id:             id.NewContext(ctx),
-		reconnecting:   new(sync.Map),
-		version:        buildVersion,
-		servicePidLock: &sync.RWMutex{},
-		servicePids:    make(map[string]map[int]string),
-		resolver:       resolver,
-		idLookup:       lookup,
+		cfg:                cfg,
+		Ctx:                ctx,
+		CancelFn:           cancel,
+		id:                 id.NewContext(ctx),
+		reconnecting:       new(sync.Map),
+		version:            buildVersion,
+		servicePidLock:     &sync.RWMutex{},
+		servicePids:        make(map[string]map[int]string),
+		resolver:           resolver,
+		idLookup:           lookup,
+		shouldIncludeEvent: sampleMatcher,
```

Comment:
it'd be nice to start cleaning up this agent context hotchpotch, could we just pass the function where it's required instead?

> out of scope: I know there'll be more deps being injected, but we probably should start considering the service injection concern using a more suitable pattern like deps composition or service locator

Reply:
Great comment, we will address it in further development.